File: java/com/example/spottio/ChatMessage.java
package com.example.spottio;

import java.util.Date;

public class ChatMessage {
    public String docId;
    public String text;
    public boolean isMe;
    public Date timestamp;
    public boolean mostraData;
    public String testoData;
    public String senderName;
    public boolean isGroup;
    public String profilePicUri;
    public boolean isVerified;

    public ChatMessage(String docId, String text, boolean isMe, Date timestamp, boolean mostraData, String testoData, String senderName, boolean isGroup, String profilePicUri, boolean isVerified) {
        this.docId = docId;
        this.text = text;
        this.isMe = isMe;
        this.timestamp = timestamp;
        this.mostraData = mostraData;
        this.testoData = testoData;
        this.senderName = senderName;
        this.isGroup = isGroup;
        this.profilePicUri = profilePicUri;
        this.isVerified = isVerified;
    }
}

File: java/com/example/spottio/ChatMessage.kt
package com.example.spottio

import java.util.Date

data class ChatMessage @JvmOverloads constructor(
    @JvmField var docId: String? = "",
    @JvmField var text: String? = "",
    @JvmField var isMe: Boolean = false,
    @JvmField var timestamp: Date? = Date(),
    @JvmField var mostraData: Boolean = false,
    @JvmField var testoData: String? = "",
    @JvmField var senderName: String? = "",
    @JvmField var isGroup: Boolean = false,

    // FIX: A prova di crash
    @JvmField var profilePicUri: String? = "",

    @JvmField var isVerified: Boolean = false
)